File: PlacementCellBackend/Services/Placements/Interfaces/ITeacherPlacementService.cs
using PlacementCellBackend.DTOs.ReasearchOpeningsDtos;

namespace PlacementCellBackend.Services.Placements.Interfaces
{
    public interface ITeacherPlacementService
    {
        Task<IEnumerable<ResearchOpeningDto>> GetAllTeacherResearchOpeningsAsync();
        Task<ResearchOpeningDto?> GetTeacherResearchOpeningByIdAsync(int id);
        Task<ResearchOpeningDto> CreateTeacherResearchOpeningAsync(ResearchOpeningCreateDto researchOpening);
        Task<bool> UpdateTeacherResearchOpeningAsync(int id, ResearchOpeningCreateDto researchOpening);
        Task<bool> DeleteTeacherResearchOpeningAsync(int id);
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/services/Placements/Interfaces/ITeacherPlacementService.java
package com.example.placementicsbackend.services.Placements.Interfaces;

import com.example.placementicsbackend.dtos.ReasearchOpeningsDtos.ResearchOpeningCreateDto;
import com.example.placementicsbackend.dtos.ReasearchOpeningsDtos.ResearchOpeningDto;
import java.util.List;

public interface ITeacherPlacementService {

    List<ResearchOpeningDto> getAllTeacherResearchOpenings();

    ResearchOpeningDto getTeacherResearchOpeningById(int id);

    ResearchOpeningDto createTeacherResearchOpening(ResearchOpeningCreateDto researchOpening);

    boolean updateTeacherResearchOpening(int id, ResearchOpeningCreateDto researchOpening);

    boolean deleteTeacherResearchOpening(int id);
}
